### PR TITLE
Enable Globally Disabling Instance Discovery Before Token Acquisition

### PR DESCRIPTION
**Pull request by msJinLei**
Usage
```powershell
Update-AzConfig -DisableInstanceDiscovery $true
Connect-AzAccount
Clear-AzConfig -DisableInstanceDiscovery
```
 Message When disable instance discovery
```
DEBUG: False MSAL 4.56.0.0 MSAL.CoreCLR .NET 8.0.1 Microsoft Windows 10.0.19045 [2024-03-11 07:56:21Z - xxxxxxxx-178d-4a69-9231-xxxxxxxxxxxx] Skipping Instance discovery for Aad authority because it is not enabled.
```

Message when enable instance disconvery
```
DEBUG: False MSAL 4.56.0.0 MSAL.CoreCLR .NET 8.0.1 Microsoft Windows 10.0.19045 [2024-03-11 08:11:44Z - xxxxxxxx-b2de-46c0-bba0-xxxxxxxxxxxx] [Instance Discovery] Instance discovery is enabled and will be performed
````

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Authentication/Config/Definitions/DisableInstanceDiscoveryConfig.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.Shared.Config;
+using Microsoft.Azure.PowerShell.Common.Config;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Config.Definitions
+{
+    /// <summary>
+    /// Definition of the config to control whether login by WAM (web account manager) or not.
```
Comment (isra-fel) [inline src/Accounts/Authentication/Config/Definitions/DisableInstanceDiscoveryConfig.cs:26]: comment is incorrect
line 30 as well

**Review by isra-fel — CHANGES_REQUESTED** [review]
Need to regen help. See https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/using-config-framework.md#step-3-regenerate-help-documents

File: src/Accounts/Authentication/Config/Definitions/DisableInstanceDiscoveryConfig.cs (hunk 1 above)
Comment (msJinLei) [reply src/Accounts/Authentication/Config/Definitions/DisableInstanceDiscoveryConfig.cs]: Done

**Comment by msJinLei** [comment]
> Need to regen help. See https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/using-config-framework.md#step-3-regenerate-help-documents

Done

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**